Description: Fix FTBFS with -Werror=format-security
 gnome-scan (0.6.2-1.1) unstable; urgency=low
 .
   * Non-maintainer upload.
   * Fix FTBFS with -Werror=format-security. (Closes: #643393).
     - Thanks Philippe De Swert for patch.
Author: Hector Oron <zumbi@debian.org>
Bug-Debian: http://bugs.debian.org/643393

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- gnome-scan-0.6.2.orig/lib/gnome-scan-module-manager.c
+++ gnome-scan-0.6.2/lib/gnome-scan-module-manager.c
@@ -200,7 +200,7 @@ gnome_scan_module_manager_query_modules
 				module = gnome_scan_module_new (filename);
 
 				if (!g_type_module_use (G_TYPE_MODULE (module))) {
-					g_warning (g_module_error ());
+					g_warning ("%s", g_module_error ());
 				}
 				manager->modules = g_slist_append (manager->modules, module);
 			}
@@ -225,7 +225,7 @@ gnome_scan_module_manager_unload_modules
 		g_type_module_unuse (G_TYPE_MODULE (node->data));
 	}
 }
-
+ 
 static gboolean
 gsmm_is_valid_module_name (const gchar*name)
 {
--- gnome-scan-0.6.2.orig/lib/gnome-scan-dialog.c
+++ gnome-scan-0.6.2/lib/gnome-scan-dialog.c
@@ -344,7 +344,7 @@ gnome_scan_dialog_dispose (GObject *obje
 	
 	if (!priv->disposed) {
 		/* freeing all scanner params */
-		if (scanner = gnome_scan_job_get_scanner(priv->job)) {
+		if ((scanner = gnome_scan_job_get_scanner(priv->job))) {
 			gnome_scan_plugin_params_foreach (GNOME_SCAN_PLUGIN(scanner),
 											  (GFunc) gsd_destroy_param,
 											  object);
@@ -358,7 +358,7 @@ gnome_scan_dialog_dispose (GObject *obje
 		}
 		
 		/* unref all sink params */
-		if (sink = gnome_scan_job_get_sink(priv->job)) {
+		if ((sink = gnome_scan_job_get_sink(priv->job))) {
 			gnome_scan_plugin_params_foreach (GNOME_SCAN_PLUGIN(sink),
 											  (GFunc) gsd_destroy_param,
 											  object);
@@ -562,9 +562,9 @@ gsd_message_dialog (GnomeScanDialog *dia
 	message_dialog = gtk_message_dialog_new (parent,
 											 GTK_DIALOG_MODAL
 											 | GTK_DIALOG_DESTROY_WITH_PARENT,
-											 mtype, btype, g_strdup (primary));
+											 mtype, btype, "%s", g_strdup (primary));
 	gtk_message_dialog_format_secondary_text (GTK_MESSAGE_DIALOG (message_dialog),
-											  g_strdup (secondary));
+											  "%s", g_strdup (secondary));
 	
 	gtk_dialog_run (GTK_DIALOG (message_dialog));
 	gtk_widget_destroy (message_dialog);
@@ -578,7 +578,7 @@ static void
 gsd_build_group_box (GnomeScanDialog *dialog, GtkWidget *page, GtkBox *box, GnomeScanPlugin *plugin, GQuark group)
 {
 	GnomeScanDialogPrivate *priv = GET_PRIVATE(dialog);
-	GtkWidget *label, *group_box, *alignment, *table, *widget, *eventbox;
+	GtkWidget *label, *group_box, *alignment, *table, *widget;
 	GSList *node = NULL;
 	const gchar *domain;
 	gboolean expands, shows_label;
@@ -1077,7 +1077,7 @@ gsd_build_scanner_ui	(GnomeScanDialog *g
 						 GNOME_SCAN_PLUGIN (scanner),
 						 GS_PARAM_GROUP_SCANNER_FRONT);
 	/* preview */
-	if (node = gnome_scan_plugin_get_param_group (GNOME_SCAN_PLUGIN (scanner), GS_PARAM_GROUP_PREVIEW)) {
+	if ((node = gnome_scan_plugin_get_param_group (GNOME_SCAN_PLUGIN (scanner), GS_PARAM_GROUP_PREVIEW))) {
 		gsd_preview_scanner_selected (gsd);
 		GTK_WIDGET_SET_VISIBLE (priv->preview_page, TRUE);
 	}
@@ -1301,7 +1301,6 @@ gsd_preview_monitor (GnomeScanDialog *gs
 static void
 gsd_preview_refresh (GtkButton *button, GnomeScanDialog *gsd)
 {
-	static GdkPoint origin = {0, 0};
 	GnomeScanDialogPrivate *priv = GET_PRIVATE (gsd);
 	GError *error = NULL;
 	GValue *value;
--- gnome-scan-0.6.2.orig/lib/gnome-scan-module.c
+++ gnome-scan-0.6.2/lib/gnome-scan-module.c
@@ -130,7 +130,7 @@ gnome_scan_module_load (GTypeModule *gmo
 	
 	module->library = g_module_open (module->filename, G_MODULE_BIND_MASK);
 	if (!module->library) {
-		g_warning (g_module_error ());
+		g_warning ("%s", g_module_error ());
 		return FALSE;
 	}
 	
